sycl/include/CL/sycl/types.hpp

```diff
@@ -273,13 +269,22 @@ template <typename T, typename R, rounding_mode roundingMode>
 detail::enable_if_t<!std::is_same<T, R>::value && is_float_to_int<T, R>::value,
                     R>
 convertImpl(T Value) {
+	R Result;
```

Comment:
These variables are used only in one case. Let's define them there.

Reply:
If we do that, we get a compilation error saying we cannot bypass the definition of this variable, or some error on those lines.

Reply:
I think you will help brackets : https://godbolt.org/z/80_anA

Reply:
Thanks. The godbolt link really helps.